src/Dma.h

```diff
@@ -61,6 +61,12 @@ namespace RTX {
   
   class Dma : public Element {
   public:
+    // Delegate class that can be subclassed to implement a different allocation demand function
+    class DemandAllocationDelegate {
+    public:
+      virtual bool allocateDemands(time_t time, Dma &dma) = 0;
```

Comment:
this is purely style/semantics, but preference is to have DMA param first, and time param second. Also please use shared pointer.